modules/log.cpp

```diff
@@ -457,12 +457,26 @@ CModule::EModRet CLogMod::OnSendToIRCMessage(CMessage& Message) {
     return CONTINUE;
 }
 
-void CLogMod::OnJoin(const CNick& Nick, CChan& Channel) {
-    if (NeedJoins()) {
-        PutLog("*** Joins: " + Nick.GetNick() + " (" + Nick.GetIdent() + "@" +
-                   Nick.GetHost() + ")",
-               Channel);
+void CLogMod::OnJoinMessage(CJoinMessage& Message) {
+    if (!NeedJoins())
+        return;
+
+    const CNick& Nick = Message.GetNick();
+    CChan& Channel = *Message.GetChan();
+    CString Account = Message.GetTag("account");
+    const char* s = " ";
+
+    if (Account.empty())
+        Account = Message.GetParam(1);
```

Comment:
Perhaps there should be some kind of `Message.GetAccount()` which handles both the tag and extended-joins. But that can wait until there's some other user of that info, for now doing it here is fine. Add a TODO though

Reply:
The issue here is that the extended-joins param is only valid for a JOIN message, so if we have a method that attempts to get it from generic messages then we need to explicitly check that it's actually a JOIN message before attempting to get extended-join info.

Reply:
Yes, this needs some consideration how to do it better, and TODO for now will be enough

Reply:
TODO added.